File: code/community/1304597127434866691/improve-speech-recognition-noisy-environments.go
package main

import (
	"context"
	"fmt"
	"os"

	"github.com/deepgram-devs/deepgram-go-client/sdk"
)

func main() {
	deepgramApiKey := os.Getenv("DEEPGRAM_API_KEY")
	audioURL := os.Getenv("AUDIO_URL")

	// Initialize Deepgram client
	client := sdk.NewClient(deepgramApiKey)

	// Prepare request options
	options := sdk.TranscriptionOptions{
		Punctuate:    true,
		NoiseReduction: true, // Enable noise reduction
	}

	// Perform transcription
	response, err := client.TranscribeURL(context.Background(), audioURL, options)
	if err != nil {
		fmt.Printf("Failed to transcribe audio: %s\n", err)
		return
	}

	// Print transcription
	fmt.Printf("Transcription: %s\n", response.Results.Channels[0].Alternatives[0].Transcript)
}

File: code/community/1304597127434866691/improve-speech-recognition-noisy-environments.cs
using System;
using System.Threading.Tasks;
using Deepgram.SDK;
using Deepgram.Transcription;

class Program
{
    static async Task Main(string[] args)
    {
        var deepgramApiKey = Environment.GetEnvironmentVariable("DEEPGRAM_API_KEY");
        var audioUrl = Environment.GetEnvironmentVariable("AUDIO_URL");

        var deepgramClient = new DeepgramClient(deepgramApiKey);

        var options = new PreRecordedTranscriptionOptions
        {
            Punctuate = true,
            NoiseFilter = true // Enable noise reduction
        };

        var response = await deepgramClient.Transcription.PrerecordedFromUrlAsync(audioUrl, options);

        Console.WriteLine($"Transcription: {response.Results.Channels[0].Alternatives[0].Transcript}");
    }
}